src/realm/util/features.h

```diff
@@ -215,6 +217,27 @@
 #define REALM_TVOS 0
 #endif
 
+// asl_log is deprecated in favor of os_log as of the following versions:
+// macos(10.12), ios(10.0), watchos(3.0), tvos(10.0)
+// versions are defined in /usr/include/Availability.h as they become available
+#if REALM_PLATFORM_APPLE \
+   && ( (defined(MAC_OS_X_VERSION_10_12) \
+         && defined(MAC_OS_X_VERSION_MIN_REQUIRED) \
+         && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12) \
```

Comment:
Is it intentional that you're using `MAC_OS_X_VERSION_MIN_REQUIRED` from the old-style AvailabilityMacros.h rather than `__MAC_OS_X_VERSION_MIN_REQUIRED` from Availability.h, as you're doing for `__IPHONE_OS_VERSION_MIN_REQUIRED` and friends?

The pattern recommended in Availability.h is to compare against literals like `101200` rather than the named macros, since that avoids the need to check whether the named macros exist.

Rather than keying off the presence of a `FOO_VERSION_BLAH` as the basis of platform detection, why not use the `REALM_IOS` and the like defined above to determine which of the Mac / iOS / tvOS / watchOS version should be checked?

Something like:
```
#if REALM_PLATFORM_APPLE \ 
    && ( \
        (REALM_IOS && __IPHONE_OS_VERSION_MIN_REQUIRED >= 100000) \
     || (REALM_TVOS && __TV_OS_VERSION_MIN_REQUIRED >= 100000) \
     || (REALM_WATCHOS && __WATCH_OS_VERSION_MIN_REQUIRED >= 30000) \
     || (__MAC_OS_X_VERSION_MIN_REQUIRED >= 101200) \
       )
```